graylog2-server/src/main/java/org/graylog2/indexer/messages/Messages.java

```diff
@@ -196,7 +213,12 @@ private void waitBeforeRetrying(long attempt) {
         }
     }
 
-    private List<IndexingError> runBulkRequest(List<IndexingRequest> indexingRequestList, int count) {
+    @SuppressWarnings("UnstableApiUsage")
+    private List<IndexingError> runBulkRequest(List<IndexingRequest> indexingRequestList, int count, IndexingListener indexingListener) {
+        final Retryer<List<IndexingError>> BULK_REQUEST_RETRYER = indexingListener == null
+                ? BULK_REQUEST_RETRYER_BUILDER.build()
+                : BULK_REQUEST_RETRYER_BUILDER.withRetryListener(retryListenerFor(indexingListener)).build();
```

Comment:
The `.withRetryListener()` modifies the `BULK_REQUEST_RETRYER_BUILDER` builder instead of returning a completely new builder instance. So every time we have a non-null `indexingListener`, we add another listener instance to the builder.

It's currently only used in tests, but the methods are public so nothing keeps us from using them in the future and then we accumulate objects. :slightly_smiling_face:

Reply:
Good catch!

Reply:
:heavy_check_mark: